apps/wallet-mobile/src/features/Receive/common/ShareQRCodeCard/ShareQRCodeCard.tsx

```diff
@@ -0,0 +1,176 @@
+  const [isSharing, setIsSharing] = React.useState<boolean>(false)
+  const strings = useStrings()
+  const ref: React.RefObject<ViewShot> = React.useRef(null)
+
+  const {styles, colors} = useStyles()
+
+  const shareImage = () => {
+    setIsSharing(true)
+  }
+
+  React.useEffect(() => {
+    if (isSharing) {
+      const captureAndShare = () => {
+        _.delay(async () => {
+          try {
+            await new Promise((resolve) => setTimeout(resolve, 50))
+
+            const uri = await captureRef(ref, {
+              format: 'png',
+              quality: 1,
+              fileName: mocks.shareFileName,
+            })
+
+            setIsSharing(false)
+            await Share.open({url: uri, filename: mocks.shareFileName, message: `${strings.address} ${address}`})
+          } finally {
+            setIsSharing(false)
+          }
+        }, 50)
+      }
+
+      captureAndShare()
+    }
+  }, [address, isSharing, strings.address])
+
+  if (isSharing)
+    return (
+      <ViewShot ref={ref}>
+        <CaptureShareQRCodeCard address={address} />
+      </ViewShot>
+    )
+
+  return (
+    <View style={styles.card}>
+      <LinearGradient
+        style={[StyleSheet.absoluteFill, {opacity: 1}]}
+        start={{x: 0, y: 0}}
+        end={{x: 0, y: 1}}
+        colors={colors.bgCard}
+      />
+
+      <Text style={styles.title}>{title}</Text>
+
+      <View style={styles.addressContainer}>
+        <View style={styles.qrCode}>
+          <QRCode value={address} size={158} backgroundColor={colors.white} color={colors.black} />
+        </View>
+
+        <Spacer height={16} />
+
+        <Text style={styles.textAddress}>{address}</Text>
+      </View>
+
+      <TouchableOpacity activeOpacity={0.5} onPress={shareImage} onLongPress={onLongPress}>
+        <Text style={styles.textShareAddress}>{strings.shareLabel}</Text>
+      </TouchableOpacity>
+
+      {isCopying && (
+        <Animated.View layout={Layout} entering={FadeInDown} exiting={FadeOutDown} style={styles.isCopying}>
+          <Text style={styles.copiedText}>{strings.addressCopiedMsg}</Text>
+        </Animated.View>
+      )}
+    </View>
+  )
+}
+
+const useStyles = () => {
+  const SCREEN_WIDTH = useWindowDimensions().width
+  const {theme} = useTheme()
+  const {color, typography} = theme
+
+  const styles = StyleSheet.create({
+    qrCode: {
+      backgroundColor: theme.color.gray.min,
+      padding: 10,
+      borderRadius: 8,
+    },
+    addressContainer: {
+      alignItems: 'center',
+    },
+    card: {
+      borderRadius: 16,
+      width: SCREEN_WIDTH - 34,
+      alignItems: 'center',
+      maxHeight: 458,
+      flex: 1,
+      minHeight: 394,
+      alignSelf: 'center',
+      overflow: 'hidden',
+      paddingVertical: 16,
+      gap: 32,
+      paddingTop: 32,
+    },
+    title: {
+      ...typography['heading-3-medium'],
+      color: color.gray.max,
+    },
+    textAddress: {
+      textAlign: 'center',
+      paddingHorizontal: 16,
```

Comment:
if it is a visual space and not part of the text, prefer a `<Spacer height={16} />` component